.travis.yml

```diff
@@ -16,16 +16,79 @@ cache:
     - $HOME/.jest-cache
     - $HOME/.npm
     - $HOME/.nvm/.cache
-    - $HOME/.phpbrew
 
 branches:
   only:
     - master
 
+env:
+  global:
+    - PUPPETEER_SKIP_CHROMIUM_DOWNLOAD: true
+    - WP_DEVELOP_DIR: ./wordpress
+    - LOCAL_SCRIPT_DEBUG: false
+    - INSTALL_COMPOSER: false
+    - INSTALL_WORDPRESS: true
+
 before_install:
   - nvm install
-
-env: PUPPETEER_SKIP_CHROMIUM_DOWNLOAD=true
+  - |
+    if [[ "$INSTALL_WORDPRESS" = "true" ]]; then
+      # Upgrade docker-compose.
+      sudo rm /usr/local/bin/docker-compose
+      curl -sL https://github.com/docker/compose/releases/download/1.24.0/docker-compose-`uname -s`-`uname -m` > docker-compose
+      chmod +x docker-compose
+      sudo mv docker-compose /usr/local/bin
+    fi
+
+install:
+  # Build Gutenberg.
+  - npm ci
+  - npm run build
+  - |
+    if [[ "$INSTALL_WORDPRESS" = "true" ]]; then
+      # Download and unpack WordPress.
+      curl -sL https://wordpress.org/nightly-builds/wordpress-latest.zip -o /tmp/wordpress-latest.zip
+      unzip -q /tmp/wordpress-latest.zip -d /tmp
+      mkdir -p wordpress/src
+      mv /tmp/wordpress/* wordpress/src
+
+      # Create the upload directory with permissions that Travis can handle.
+      mkdir -p wordpress/src/wp-content/uploads
+      chmod 767 wordpress/src/wp-content/uploads
+
+      # Grab the tools we need for WordPress' local-env.
+      curl -sL https://github.com/WordPress/wordpress-develop/archive/master.zip -o /tmp/wordpress-develop.zip
+      unzip -q /tmp/wordpress-develop.zip -d /tmp
+      mv \
+        /tmp/wordpress-develop-master/tools \
+        /tmp/wordpress-develop-master/tests \
+        /tmp/wordpress-develop-master/.env \
+        /tmp/wordpress-develop-master/docker-compose.yml \
+        /tmp/wordpress-develop-master/wp-cli.yml \
+        /tmp/wordpress-develop-master/*config-sample.php \
+        /tmp/wordpress-develop-master/package.json wordpress
+
+      # Install WordPress.
+      cd wordpress
+      npm install dotenv wait-on
```

Comment:
Not sure I understand the purpose or need for these installs.

`wait-on` seems like something which could have been useful in place of the `sleep 10` which follows a few lines down. I suppose it might have been an initial attempt to try to react to the availability of the environment in a more reliable way than a `sleep`?

Reply:
#18924